redbot/cogs/warnings/warnings.py

```diff
@@ -342,36 +341,48 @@ def __init__(self, bot: Red):
 
     @commands.command()
     @commands.guild_only()
+    @checks.admin()
     async def warnings(
         self, ctx: commands.Context, user: Optional[Union[discord.Member, int]] = None
     ):
-        """List the warnings for the specified user.
-
-        Omit `<user>` to see your own warnings.
-
-        Note that showing warnings for users other than yourself requires
-        appropriate permissions.
-        """
+        """List the warnings for the specified user."""
         if user is None:
-            user = ctx.author
-        else:
-            if not await is_admin_or_superior(self.bot, ctx.author):
-                return await ctx.send(
-                    warning(_("You are not allowed to check warnings for other users!"))
+            await ctx.send(_("Please provide a user you want to check the warnings for."))
+
+        msg = ""
+        member_settings = self.config.member(user)
```

Comment:
This doesn't properly account for the possibility a user id was passed without it being to a current user (and needs to)

Reply:
Honestly I have no idea how to go about this.

Reply:
The removed handling here was doing that before:
https://github.com/Cog-Creators/Red-DiscordBot/pull/3327/files#diff-bd66023271557007376afcf648f3b3ffL363-L368